runtime/kubernetes.go

```diff
@@ -34,74 +38,580 @@ type Kubernetes struct {
+}
+
+type Rollbackspec struct {
+	ApiVersion         string            `json:"apiVersion"`
+	Name               string            `json:"name"`
+	UpdatedAnnotations map[string]string `json:"updatedAnnotations,omitempty"`
+	RollbackTo         Revisionspec      `json:"rollbackTo"`
+}
+
+type JobScaleGet struct {
+	Kind       string `json:"kind"`
+	APIVersion string `json:"apiVersion"`
+	Metadata   struct {
+		Name              string    `json:"name"`
+		Namespace         string    `json:"namespace"`
+		SelfLink          string    `json:"selfLink"`
+		UID               string    `json:"uid"`
+		ResourceVersion   string    `json:"resourceVersion"`
+		CreationTimestamp time.Time `json:"creationTimestamp"`
+		Labels            struct {
+			Name  string `json:"name"`
+			Space string `json:"space"`
+		} `json:"labels"`
+	} `json:"metadata"`
+	Spec struct {
+		Parallelism           int `json:"parallelism"`
+		Completions           int `json:"completions"`
+		ActiveDeadlineSeconds int `json:"activeDeadlineSeconds"`
+		Selector              struct {
+			MatchLabels struct {
+				ControllerUID string `json:"controller-uid"`
+			} `json:"matchLabels"`
+		} `json:"selector"`
+		Template struct {
+			Metadata struct {
+				Name              string      `json:"name"`
+				Namespace         string      `json:"namespace"`
+				CreationTimestamp interface{} `json:"creationTimestamp"`
+				Labels            struct {
+					ControllerUID string `json:"controller-uid"`
+					JobName       string `json:"job-name"`
+					Name          string `json:"name"`
+					Space         string `json:"space"`
+				} `json:"labels"`
+			} `json:"metadata"`
+			Spec struct {
+				Containers []struct {
+					Name  string `json:"name"`
+					Image string `json:"image"`
+					Env   []struct {
+						Name  string `json:"name"`
+						Value string `json:"value"`
+					} `json:"env"`
+					TerminationMessagePath   string `json:"terminationMessagePath"`
+					TerminationMessagePolicy string `json:"terminationMessagePolicy"`
+					ImagePullPolicy          string `json:"imagePullPolicy,omitempty"`
+				} `json:"containers"`
+				RestartPolicy                 string `json:"restartPolicy"`
+				TerminationGracePeriodSeconds int    `json:"terminationGracePeriodSeconds"`
+				DNSPolicy                     string `json:"dnsPolicy"`
+				SecurityContext               struct {
+				} `json:"securityContext"`
+				ImagePullSecrets []struct {
+					Name string `json:"name"`
+				} `json:"imagePullSecrets,omitempty"`
+				SchedulerName string `json:"schedulerName"`
+			} `json:"spec"`
+		} `json:"template"`
+	} `json:"spec"`
+}
+
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
 	}
-	if config.AuthVaultPath == "" {
-		log.Fatalln("No kubernetes auth vault path was available! Aborting.")
+	return os.Getenv("USERPROFILE") // windows
+}
+
+func NewKubernetes(name string, imagePullSecret string) (r Runtime) {
+	// Check if we have a service account in cluster
```

Comment:
Rather than pull from the database and coalesce our own client, use the kubeconfig from a service account or `--kubeconfig` or from `$HOME/.kube/config` to determine kubernetes cluster.